src/content/blog/future-of-astro-zero-js-view-transitions.mdx

```diff
@@ -0,0 +1,106 @@
+socialImage: "/src/content/blog/_images/future-of-astro-zero-js-view-transitions/og-astro-zerojs-transitions.webp"
+lang: "en"
+---
+
+import BlogSummary from "/src/components/BlogSummary.astro";
+
+<BlogSummary>
+  This is Part 1 of our series on *The Future of Astro* covering three major new features we have planned for Astro in 2024. This post introduces an update to the [View Transitions API](https://docs.astro.build/en/guides/view-transitions/) that Astro can now leverage for native, app-like page navigation without a single line of JavaScript required.
+</BlogSummary>
+
+The [View Transitions API](https://developer.chrome.com/docs/web-platform/view-transitions) is a set of new platform APIs that unlock native browser transitions between pages. Historically this has only been possible in JavaScript-heavy Single Page Applications (SPAs), but recent advances are now bringing native page transitions to the web platform.
+
+Astro users have had early access to view transitions since [Astro 2.9](https://astro.build/blog/astro-290/). Due to initial limitations in the browser, our early support worked by injecting a JavaScript router onto the page to manage the animations for you. This was never meant as a long-term solution, but it successfully created an early playground for our users to explore:
+
+<div class="flex flex-col gap-6 my-10 items-end">
+	<figure>
+		<video autoplay loop muted playsinline class="border border-neutral-600 rounded">
+			<source src="/assets/blog/astro-3/demo-2.mp4" type="video/mp4" />
+			Sorry, your browser doesn't support embedded videos.
+		</video>
+		<figcaption class="text-center text-sm my-4">
+			Watch <a href="https://twitter.com/joebell_/status/1688167865961037825">Joe Bell</a> demo view transitions.
+		</figcaption>
+	</figure>
+	<figure>
+		<video autoplay loop muted playsinline class="border border-neutral-600 rounded">
+			<source src="/assets/blog/astro-3/demo-3.mp4" type="video/mp4" />
+			Sorry, your browser doesn't support embedded videos.
+		</video>
+		<figcaption class="text-center text-sm my-4">
+			Try out this <a href="https://astro-records.pages.dev/">live demo</a> by <a href="https://twitter.com/charca">Maxi Ferreira</a>.
+		</figcaption>
+	</figure>
+</div>
+
+Our bet was that the major web browsers would continue to improve and one day we could remove our JavaScript router shim entirely. One day, when a user clicked a link, that click could trigger a transition animation in the browser as part of its normal flow from page to page.
+
+I'm excited to announce that that day has finally arrived. Chrome 126 (released last week) and Microsoft Edge 126 (releasing this week) both support "cross-document view transitions" unlocking **zero-JavaScript view transitions** in Astro.
+
+## Demo
+
+Both Chrome and Edge offer staggered rollout of new versions, so check that your browser is running 126 before running the demo. In chrome, you can see your current version at `chrome://settings/help`.
```

Comment:
Chrome's stable will be June 25th, not sure if that's worth mentioning or not: https://chromiumdash.appspot.com/schedule

Reply:
<img width="396" alt="Screen Shot 2024-06-11 at 3 31 15 PM" src="https://github.com/withastro/astro.build/assets/622227/9c24170b-f9da-4883-b20a-828a5d300d2b">

That's Chrome OS! Chrome Browser was technically last week but then today is the first day that most people get access via auto-upgrade rollout behavior.

Reply:
But I'll add a note that auto-upgrading may not be enough to get access.

Reply:
Oops, my bad. Ok, I just noticed that I haven't received it yet and figured it was the gradual rollout.

Reply:
no problem, I had the same thought. Rewritten to be more explicit on this